discovery_engine_core/bindings/src/types/document/document.rs

```diff
@@ -18,7 +18,28 @@ use std::ptr::addr_of_mut;
 
 use uuid::Uuid;
 
-use xayn_discovery_engine_core::document::{Document, Embedding, NewsResource, UserReaction};
+use xayn_discovery_engine_core::{
+    document::{Id, NewsResource, UserReaction},
+    stack::Id as StackId,
+};
+
+pub struct Document {
```

Comment:
:grey_question: why is it beneficial to introduce this here?

Reply:
the `core::document::Document` has the embedding field, which is not used at the dart side anymore. hence i introduced this new type to only transfer the required fields. since this is a ffi detail, i added it to `bindings` instead of `core`.